docs/developing/http-apis.md

```diff
@@ -12,48 +13,147 @@ useful for:
 
 * Many other things!
 
-All APIs for all apps on a particular Sandstorm server are served from
-the same hostname -- for instance, the Sandstorm Alpha server serves
-APIs from https://alpha-api.sandstorm.io. The specific API to which a
-request is addressed is determined by examining the `Authorization`
-header. The header must be of the form:
+## Overview
 
-    Authorization: Bearer <token>
+When custom code needs to interact with a Sandstorm app, it sends a
+HTTP request to the **API hostname** of the Sandstorm host where app
+is running, along with an **API token** embedded in an `Authorization`
+HTTP header.
 
-The specific value of `<token>` both determines the specific app
-instance to which the request is addressed as well as authorizes the
-caller to send requests to that app.
+You can try making a request to a Sandstorm app's API right now via `curl`:
 
-Because an `Authorization` header is required, it is impossible for a
-web browser to open a Sandstorm HTTP API directly in a browser
-window. This is intentional: this prevents Sandstorm apps from
-executing arbitrary scripts from the API host.
+```bash
+curl -H "Authorization: Bearer 49Np9sqkYV4g_FpOQk1p0j1yJlvoHrZm9SVhQt7H2-9" https://alpha-api.sandstorm.io/
+```
 
-There are various ways to obtain an API key:
+Sandstorm is responsible for generating the API token. The API token
+is used for both **access control** and **routing a request** to the
+appropriate grain.
 
-* The user can click the key icon in the top bar when they have an app
-  open.
+When a request comes in with a valid API token, Sandstorm sanitizes
+the request, removing the `Cookie` header and the API token, adding
+typical Sandstorm authentication headers like `X-Sandstorm-User-Id`,
+and passes the request to the app. Sandstorm combines it with the app
+package's `bridgeConfig.apiPath` as part of sending the request to the
+grain. Sandstorm sanitizes responses and removes any `Set-Cookie`
+response header.
+
+The API endpoint is set up to allow **cross-origin requests from any
+origin**, which means you can access an API from `XMLHttpRequest` on
+any domain.
+
+An app can **request the generation of an API token**, and a Sandstorm
+user can manually create a valid API token by clicking on the Webkey icon
+in the Sandstorm shell.
+
+## How to generate an API token
+
+There are various ways to obtain an API token:
 
-* The app itself can make a call to
+* The best is via *offer templates*, where the app specifies textual
+  information for the user of the app, and Sandstorm places the token
+  into this template before displaying it to the user.
+
+* The app itself can make a Cap'n Proto RPC call to
```

Comment:
Consider indicating that this is deprecated and is slated for removal in future versions of Sandstorm, to strongly encourage people to avoid this option.  Also, consider moving it to the end of the list, to indicate its inferiority.